### PR TITLE
Updating tests to use moq.netcore

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @rynowak

File: test/Microsoft.AspNet.Routing.Tests/Constraints/RegexInlineRouteConstraintTests.cs [hunk 1]
```diff
@@ -59,36 +58,23 @@ public void RegexInlineConstraint_IsCultureInsensitive()
...
-
-            // Assert
-            Assert.False(matchInUsEnglish); // this just verifies the test
-            Assert.False(matchInTurkish);
         }
 
-        private static bool EasyMatch(IRouteConstraint constraint,
-                                      string routeKey,
-                                      RouteValueDictionary values)
+        private static bool EasyMatch(
```
Comment (rynowak) [inline test/Microsoft.AspNet.Routing.Tests/Constraints/RegexInlineRouteConstraintTests.cs]: Ugh, this is annoying.

File: test/Microsoft.AspNet.Routing.Tests/Template/TemplateBinderTests.cs [hunk 2]
```diff
@@ -686,405 +684,6 @@ public void GetUrlWithCatchAllWithNullValue()
                 "v1");
         }
 
-#if ROUTE_COLLECTION
```
Comment (rynowak) [inline test/Microsoft.AspNet.Routing.Tests/Template/TemplateBinderTests.cs]: Wait, why did you delete these? I thought you were removing the ifdef

Comment (pranavkm) [inline test/Microsoft.AspNet.Routing.Tests/Template/TemplateBinderTests.cs]: Oh, I thought you said it was ok to remove it. I can revive it.

**Comment by pranavkm** [comment]
:up: :date:

File: src/Microsoft.AspNet.Routing/Constraints/RegexRouteConstraint.cs [hunk 3]
```diff
@@ -42,7 +42,7 @@ public RegexRouteConstraint(string regexPattern)
             HttpContext httpContext,
             IRouter route,
             string routeKey,
-            IDictionary<string, object> routeValues,
+            IDictionary<string, object> values,
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Routing/Constraints/RegexRouteConstraint.cs]: Matching `IRouteConstraint`

**Comment by rynowak** [comment]
:shipit:
